book-2-mvc/chapters/INTRO_TO_MVC.md

```diff
@@ -0,0 +1,126 @@
+
+```sql
+CREATE TABLE Animal (
+    Id	integer NOT NULL PRIMARY KEY IDENTITY,
+    [Name]	varchar(80) NOT NULL,
+    Breed	varchar(80),
+    OwnerId	integer NOT NULL,
+    CONSTRAINT FK_AnimalOwner FOREIGN KEY(OwnerId) REFERENCES [Owner](Id)
+);
+```
+
+This table stores the raw data for animals. When you query the database for an animal, you want to represent that data as an object in your code. Below is the most basic representation of that database table as a class.
+
+
+```cs
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+
+namespace DogWalker.Models
+{
+    public class Dog
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+        public string Breed { get; set; }
+        public int OwnerId { get; set; }
+        public Owner Owner { get; set; }
+    }
+}
+```
+
+Something to note here.. 
+
+In addition to properties that are foriegn keys like `OwnerId`, we may also want to include a property for the entire foriegn entity. That is why there is a property for the Owner object.
+
+### Views
+
+In MVC, the View is sometimes refered to as part of the Presentation Layer. It is an html template, similar to the JSX code you are used to writing in react, except it has the power of C# instead of JS. If it is given the context of a C# object, it knows how to dynamically create HTML.
+
+```html+razor
+@model List<Kennel.Models.Animal>
+
+<h1>All The Animals</h2>
+@foreach(Animal animal in Model)
+{
+    <div class="animal__card">
+        <h3>@animal.Name</h3>
+        <h5>Breed: @animal.Breed</h5>
+        <span>Animal is owned by @animal.Owner.Name</span>
+    </div>
+}
+```
+
+The First line of this code `@model List<Kennel.Models.Animal>` is telling the view what _type_ of object is going to be passed to it. In this case, the view is going to be passed a List of Animal objects. When the server has to respond a request for the homepage, the view will iterate over of the list of animals and create a div element for each one. The html that gets generated will be what the server sends back to a user's browser.
+
+### Controllers
+
+First, controllers are just classes.
+
+Second, the job of these controller classes is to handle incoming HTTP requests. 
+
+The controller has all of the logic needed to generate the  response requested by the client. It can validate the user, validate request data, query the database, contain logic for manipulating the data, and generate an HTML response.
+
+```csharp
+public class AnimalsController : Controller
+{
+    public ActionResult Index()
+    {
+        AnimalRepository repo = new AnimalRepository();
+        List<Animal> animals = repo.GetAllAnimals();
+
+        return View(animals);
+    }
+}
+```
+
+In this code, the controller is simply getting all the animals from the database and handing it to the View. 
+
+
+
+![](./images/asp-net-workflow.png)
```

Comment:
I think this picture should be near the top where we talk about the MVC flow. It would also be good if we could find an image that illustrates a react/api app.